Build failure: `python3Packages.sqlalchemy-continuum`

### Steps To Reproduce

```console
$ nix build nixpkgs#python3Packages.sqlalchemy-continuum
```

### Build log

```console
Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.10.9, pytest-7.2.0, pluggy-1.0.0
rootdir: /build/SQLAlchemy-Continuum-1.3.14
collecting ... ^Mcollecting 0 items                                                             ^Mcollecting 276 items                                                 >

==================================== ERRORS ====================================
_________________ ERROR collecting tests/plugins/test_flask.py _________________
ImportError while importing test module '/build/SQLAlchemy-Continuum-1.3.14/tests/plugins/test_flask.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/0pyymzxf7n0fzpaqnvwv92ab72v3jq8d-python3-3.10.9/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/plugins/test_flask.py:5: in <module>
    from flask_sqlalchemy import SQLAlchemy, _SessionSignalEvents
E   ImportError: cannot import name '_SessionSignalEvents' from 'flask_sqlalchemy' (/nix/store/1pkvi1vy9hxkngbva3jy9pz7kh0z3h3k-python3.10-Flask-SQLAlchemy-3.0.2/lib/p>
=============================== warnings summary ===============================
sqlalchemy_continuum/plugins/flask.py:25
  /build/SQLAlchemy-Continuum-1.3.14/sqlalchemy_continuum/plugins/flask.py:25: DeprecationWarning: '_app_ctx_stack' is deprecated and will be remoevd in Flask 2.3.
    from flask.globals import _app_ctx_stack, _request_ctx_stack

sqlalchemy_continuum/plugins/flask.py:25
  /build/SQLAlchemy-Continuum-1.3.14/sqlalchemy_continuum/plugins/flask.py:25: DeprecationWarning: '_request_ctx_stack' is deprecated and will be remoevd in Flask 2.3.
    from flask.globals import _app_ctx_stack, _request_ctx_stack

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR tests/plugins/test_flask.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
========================= 2 warnings, 1 error in 2.37s =========================
/nix/store/b09v23lirgvci3wzszh22mbkdfj0h0yq-stdenv-linux/setup: line 1582: pop_var_context: head of shell_variables not a function context

```

### Additional context

This also breaks the build of `python3Packages.ihatemoney` which relies on this package.

### Notify maintainers

@symphorien (`ihatemoney`)

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.10, NixOS, 23.05 (Stoat), 23.05.20230215.6d33e5e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.2`
 - channels(qaristote): `""`
 - channels(root): `"home-manager, nixos, nixos-hardware, nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/development/python-modules/flask-sqlalchemy/default.nix
+++ b/pkgs/development/python-modules/flask-sqlalchemy/default.nix
@@ -6,17 +6,20 @@
 , mock
 , sqlalchemy
 , pytestCheckHook
+, pythonOlder
 }:
 
 buildPythonPackage rec {
   pname = "flask-sqlalchemy";
-  version = "3.0.2";
+  version = "3.0.3";
   format = "pyproject";
+
+  disabled = pythonOlder "3.9";
 
   src = fetchPypi {
     pname = "Flask-SQLAlchemy";
     inherit version;
-    hash = "sha256-FhmfWz3ftp4N8vUq5Mdq7b/sgjRiNJ2rshobLgorZek=";
+    hash = "sha256-J2QzXzydfr3J7WBEr6+Yqun6UNegdM71Xd4wfslZA+w=";
   };
 
   nativeBuildInputs = [
@@ -47,7 +50,7 @@ buildPythonPackage rec {
   meta = with lib; {
     description = "SQLAlchemy extension for Flask";
     homepage = "http://flask-sqlalchemy.pocoo.org/";
-    changelog = "https://github.com/pallets-eco/flask-sqlalchemy/blob${version}/CHANGES.rst";
+    changelog = "https://github.com/pallets-eco/flask-sqlalchemy/blob/${version}/CHANGES.rst";
     license = licenses.bsd3;
     maintainers = with maintainers; [ gerschtli ];
   };

--- a/pkgs/development/python-modules/flask-sqlalchemy/default.nix
+++ b/pkgs/development/python-modules/flask-sqlalchemy/default.nix
@@ -9,12 +9,13 @@
 }:
 
 buildPythonPackage rec {
-  pname = "Flask-SQLAlchemy";
+  pname = "flask-sqlalchemy";
   version = "3.0.2";
   format = "pyproject";
 
   src = fetchPypi {
-    inherit pname version;
+    pname = "Flask-SQLAlchemy";
+    inherit version;
     hash = "sha256-FhmfWz3ftp4N8vUq5Mdq7b/sgjRiNJ2rshobLgorZek=";
   };
 
@@ -39,9 +40,14 @@ buildPythonPackage rec {
     "test_persist_selectable"
   ];
 
+  pythonImportsCheck = [
+    "flask_sqlalchemy"
+  ];
+
   meta = with lib; {
     description = "SQLAlchemy extension for Flask";
     homepage = "http://flask-sqlalchemy.pocoo.org/";
+    changelog = "https://github.com/pallets-eco/flask-sqlalchemy/blob${version}/CHANGES.rst";
     license = licenses.bsd3;
     maintainers = with maintainers; [ gerschtli ];
   };

--- a/pkgs/development/python-modules/sqlalchemy-continuum/default.nix
+++ b/pkgs/development/python-modules/sqlalchemy-continuum/default.nix
@@ -54,14 +54,15 @@ buildPythonPackage rec {
     psycopg2
     pymysql
     pytestCheckHook
-  ] ++ passthru.optional-dependencies.flask
-  ++ passthru.optional-dependencies.flask-login
-  ++ passthru.optional-dependencies.flask-sqlalchemy
-  ++ passthru.optional-dependencies.flexmock
-  ++ passthru.optional-dependencies.i18n;
+  ] ++ lib.flatten (builtins.attrValues passthru.optional-dependencies);
 
-  # indicate tests that we don't have a database server at hand
+  # Indicate tests that we don't have a database server at hand
   DB = "sqlite";
+
+  disabledTestPaths = [
+    # Test doesn't support latest SQLAlchemy
+    "tests/plugins/test_flask.py"
+  ];
 
   pythonImportsCheck = [
     "sqlalchemy_continuum"

